### DISCUSSION TITLE
Is it possible to use NetworkStream.WriteAsync(ReadOnlyMemory<byte>) under .NET Standard 2.0?

### DISCUSSION THREAD

**Discussion by GF-Huang** [dotnet/runtime#47978] [Q&A] [answered]
Is there any nuget packages contains something stream extensions to do this?

**Comment by huoyaoyuan** [comment, marked as answer]
Packages can provide additional types, but can't extend members on existing type.

https://github.com/dotnet/runtime/blob/1fe9cbff8c38592221fcdf8167c89e1b78709bd0/src/libraries/System.Private.CoreLib/src/System/IO/Stream.cs#L609-L619

You can copy the default implementation in `Stream` to get a suboptimal implementation.

**Reply by GF-Huang** [reply]
This seems to be simple wrapper and one more redundant copy operation. 

And I read the source of `NetworkStream`, it looks like pass the `ReadOnlyMemory` to the socket directly, not a wrapper.

https://source.dot.net/#System.Net.Sockets/System/Net/Sockets/NetworkStream.cs,e3c427161b014da6

![image](https://user-images.githubusercontent.com/4510984/107225413-b9627780-6a53-11eb-8a61-d84423dfbaa6.png)

**Reply by huoyaoyuan** [reply]
On netstandard2.0 you have no way to reduce the redundant copy. The BCL implementation was not that extensible.
